OpenRA.Mods.Common/Traits/World/ShroudRenderer.cs

```diff
@@ -159,6 +176,61 @@ public ShroudRenderer(World world, ShroudRendererInfo info)
 			notVisibleEdges = info.UseExtendedIndex ? Edges.AllSides : Edges.AllCorners;
 		}
 
+		Sheet CopySpritesOntoSingleSheet(out SheetBuilder sheetBuilder)
```

Comment:
Could you move this sheet management into your VertexCache or another helper class?